### PR TITLE
added a /docs folder, and caveats.md

### PR DESCRIPTION
**Pull request by cblgh**
purpose of caveats.md: documenting various gotchas that arise when people write netsim tests that do not perform the way they expect, especially when the reason for the behaviour is outside of netsim's domain

changed the internal links in various markdown docs to redirect to the docs folder with `sed`; hoping that worked well!!

### CODE REVIEW

**Comment by cblgh** [comment]
cc @cryptix do you feel these are fair wrt go-ssb? i mainly want to surface some of the gotchas / unexpected behaviour people have run into when netsimming, and suggest some steps forward :~

File: docs/caveats.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+node - access denied
+```
+
+The reason is that the go server is strict with who it allows to establish connections with it.
+If the peer running the go server does not follow the connecting party, then the connection
+will be denied.
+
+**Mitigations:**
+
+1. Make sure `alice` follows `gopher` (and observe the 3 second caveat mentioned elsewhere
```
Comment (cryptix) [inline docs/caveats.md:80]:
```suggestion
1. Make sure `alice` follows `gopher` and `gopher` follows `alice` (and observe the 3 second caveat mentioned above)
```

**Review by cryptix — APPROVED** [review]
Thanks for putting this together. These are all correct AFAICT.

As mentioned in the call we had earlier, would be good to add that these mitigations are added to the shim.sh (already?)

**Comment by cblgh** [comment]
nice :) then i'll put one last push into this & merge it!

File: docs/caveats.md (hunk 1 above)
Comment (cblgh) [reply docs/caveats.md]: added in 483bdc3

**Pull request merged**